File: samples/scripts/selection/checkProperties.py
# Script using property file and two logical conditions

from org.ow2.proactive.scripting.helper.selection import SelectionUtils
from org.ow2.proactive.scripting.helper.selection import Condition
from org.ow2.proactive.scripting.helper.selection import Conditions

#-- We set our 2 conditions by creating "Condition" objects
condition1 = Condition("ram", SelectionUtils.GREATER_THAN, "1024");
condition2 = Condition("architecture", SelectionUtils.CONTAINS, "6");

#-- We create a list of "Condition" objects
conditions = Conditions();

#-- and insert our 2 objects
conditions.add(condition1);
conditions.add(condition2);

#-- Evaluation by calling CheckConfig method
if SelectionUtils.checkProperties("samples/scripts/selection/samplePropertiesFile.txt", conditions):
    selected = True;
else:
    selected = False;

File: samples/scripts/selection/checkProperties.js
// Script using property file and two logical conditions

/* We set our 2 conditions by creating "Condition" objects */
var condition1 = new org.ow2.proactive.scripting.helper.selection.Condition("ram",org.ow2.proactive.scripting.helper.selection.SelectionUtils.GREATER_THAN, "1024");
var condition2 = new org.ow2.proactive.scripting.helper.selection.Condition("architecture",org.ow2.proactive.scripting.helper.selection.SelectionUtils.CONTAINS, "6");

/* We create a list of "Condition" objects */
var conditions = new org.ow2.proactive.scripting.helper.selection.Conditions();

/* and insert our 2 objects */
conditions.add(condition1);
conditions.add(condition2);

/* Evaluation by calling CheckConfig method */
if (org.ow2.proactive.scripting.helper.selection.SelectionUtils.checkProperties("samples/scripts/selection/samplePropertiesFile.txt", conditions)) {
    selected = true;
} else {
    selected = false;
}

